internal/updater/updater.go

```
@@ -0,0 +1,244 @@
+
+	fmt.Printf("Successfully updated to version %s\n", updateInfo.LatestVersion.String())
+	return nil
+}
+
+// createBackup creates a backup of the current executable
+func (u *Updater) createBackup(backupPath string) error {
+	sourceFile, err := os.Open(u.executablePath)
+	if err != nil {
+		return fmt.Errorf("failed to open source file: %w", err)
+	}
+	defer sourceFile.Close()
```

golangci-lint (1.64.8)

Error return value of `sourceFile.Close` is not checked

(errcheck)